diff --git a/Jakefile.js b/Jakefile.js
index 16786026..507a6df1 100644
--- a/Jakefile.js
+++ b/Jakefile.js
@@ -41,12 +41,11 @@ var webclient = path.join(web,"client");
 //
 // > jake compiler VARIANT=release
 
-var kokaDir   = "../koka-0.6"
-var libraryDir= path.join(kokaDir,"lib")
-var kokaExe   = path.join(kokaDir,"out/release/koka-0.6.0-dev")
+var kokaDir   = "/usr/local/lib/koka-0.6-dev"
+var kokaExe   = "koka-0.6.0-dev"
 var testDir   = "test";
 
-var kokaFlags = "-i" + sourceDir + " -i" + libraryDir + " " + (process.env.kokaFlags || "");
+var kokaFlags = "-i" + sourceDir + " " + (process.env.kokaFlags || "");
 var kokaCmd = kokaExe + " " + kokaFlags + " -c -o" + outputDir + " --outname=" + main + " " 
 
 
